import "common-proto.proto";

package diamond.frontend.proto;

message GetMessage {
    required uint64 clientid = 1;
    required uint32 msgid = 2;
    required uint64 txnid = 3;
    repeated string keys = 4;
    optional uint64 timestamp = 5;
}

message GetReply {
    required int32 status = 1;
    required uint32 msgid = 2;
    repeated ReadReply replies = 3;
}

message CommitMessage {
    required uint64 clientid = 1;
    required uint32 msgid = 2;
    required uint64 txnid = 3;
    required TransactionMessage txn = 5;
}

message CommitReply {
    required int32 status = 1;
    required uint64 txnid = 2;
    required uint32 msgid = 3;
    required uint64 timestamp = 4;  
}

message AbortMessage {
    required uint64 clientid = 1;
    required uint64 txnid = 2;
    required uint32 msgid = 3;
    required TransactionMessage txn = 4;
}

message AbortReply {
    required int32 status = 1;
    required uint64 txnid = 2;
    required uint32 msgid = 3;
}
